rootfs/etc/haproxy/template/haproxy.tmpl

```diff
@@ -135,6 +141,9 @@ frontend httpfront
 {{ end }}
 {{ range $location := $server.Locations }}
     use_backend {{ $location.Backend }} if ishost-{{ labelize $server.Hostname }} { path_beg {{ $location.Path }} }
+{{ if ne $server.Alias "" }}
+    use_backend {{ $location.Backend }} if ishost-{{ labelize $server.Hostname }}-alias { path_beg {{ $location.Path }} }
+{{ end }}
```

Comment:
Create a new iteration in order to move all alias to the bottom. This will ensure that hostnames will match instead of a regex.